cmd/zpool/zpool_vdev.c

```diff
@@ -721,8 +722,16 @@ make_leaf_vdev(nvlist_t *props, const char *arg, uint64_t is_log)
 		char *value = NULL;
 
 		if (nvlist_lookup_string(props,
-		    zpool_prop_to_name(ZPOOL_PROP_ASHIFT), &value) == 0)
+		    zpool_prop_to_name(ZPOOL_PROP_ASHIFT), &value) == 0) {
 			zfs_nicestrtonum(NULL, value, &ashift);
+			if (ashift != 0 &&
+			    (ashift < ASHIFT_MIN || ashift > ASHIFT_MAX)) {
+				(void) fprintf(stderr, dgettext(TEXT_DOMAIN,
+				    "property 'ashift' value %" PRIu64 " is "
+				    "invalid.\n"), ashift);
```

Comment:
I'll also update `zpool_valid_proplist()` error message adding "_only values between $ASHIFT_MIN and $ASHIFT_MAX are allowed_"